content/docs/guides/revocation.md

```diff
@@ -14,6 +14,119 @@ toc = false
+so is the stateless design impossible after all? Shouldn't we go back to stateful systems?
+
+If the architecture we are designing can rely on a central state, it will be the simplest
+approach and probably the right solution. But there are good reasons to choose the stateless
+design:
+- **scaling**: a central authorization service that is queried on every request is a single point
+of failure for the entire system. If it is down, nobody can log in, but existing sessions will
+fail too. A stateless approach will decouple session creation from authorization, so existing
+sessions can still work when the authentication service fails.
+- **isolation**: the service receiving the request might be less trusted and should not be
+able to access session information.
+- **authentication delegation**: authentication could be in a separate service (example: SSO)
+that can't be queried on every request. That service could even be managed by a different
+company.
+
+In those cases, separating authorization from session creation makes sense, but then how do
+we close a session? It is usually done through token revocation: the authorizer needs to
+know a list of tokens that must be refused, and that list changes dynamically, so we are
+reintroducing some state in the system.
+
+But revocation has properties that make it nice to implement in stateless architectures:
+- we do not need to know about all of the tokens, only those that were revoked, which will
+be much smaller
+- the list of revoked tokens will naturally grow, but if tokens have an expiration date, they
+can be purged from the list after a while
+- it is read-oriented and highly cacheable: once a token was added to the revocation list,
+we won't modify its entry (except when purging), so we don't need synchronization or consensus
+- revocation lists do not hold any critical or private information, they can be shared with
+every service
+
+So handling revocation is adding some shared state, but much more limited than what we would
+have with a fully centralized architecture.
+
+## How to implement revocation in our infrastructure?
+
+We need a reliable way to transmit revocation information to services. That will depend on
+how quickly we want to disseminate it, and how much complexity we can bear.
+
+### The basic solution: download the revocation list at startup
+
+In some cases, like communication betwen automated services, revocation is rare, mostly when
+a service is stopped, so the revocation list is mostly static and small. If we can accept some
+delay in synchronization, we can have services download the revocation list at startup. They
+will check tokens from an in memory list, that will stay the same for the entire life of the
+service.
+
+The tradeoff here is that if we need to revoke a token urgently, we will need to redeploy
+a lot of services at once.
+
+Since the revocation list is small and static, it can be stored as a file in an object
```

Comment:
```suggestion
In the case where more services accept tokens, it will become necessary to have a centrally defined list that is then distributed to all services.
Since the revocation list is small and static, it can be stored as a file in an object
```